src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -89,7 +89,7 @@ def interpreter_constraints(self) -> tuple[str, ...]:
             #  We'll probably want to find and modify all those tests to set an explicit IC, but
             #  that will take time.
             if "PYTEST_CURRENT_TEST" in os.environ:
-                return (">=3.7,<4",)
+                return (">=3.8,<4",)
```

Comment:
Tangent: There are *so many places* where we set the interpreter constraints.  This one took me a while to find.

Reply:
Maybe we should have a constant `MINIMUM_SUPPORTED_PYTHON_VERSION` instead of hard coding?

Reply:
Yeah reasonable followup. (Or rather we should have *fewer* such constants.)